// ngx-logger/esm2015/lib/config.engine.js
import { LoggerConfig } from './logger.config';
export class NGXLoggerConfigEngine {
    constructor(config) {
        this.config = config;
        this._config = config;
    }
    /** Get a readonly access to the level configured for the NGXLogger */
    get level() {
        return this._config.level;
    }
    /** Get a readonly access to the serverLogLevel configured for the NGXLogger */
    get serverLogLevel() {
        return this._config.serverLogLevel;
    }
    updateConfig(config) {
        this._config = this._clone(config);
    }
    getConfig() {
        return this._clone(this._config);
    }
    // TODO: add tests around cloning the config. updating an object passed into the config (or retrieving from the config)
    // should not update the active config, this is a shallow clone. If our config ever becomes hierarchical we must make
    // this a deep clone
    _clone(object) {
        const cloneConfig = new LoggerConfig();
        Object.keys(object).forEach((key) => {
            cloneConfig[key] = object[key];
        });
        return cloneConfig;
    }
}
